### ISSUE TITLE
simulating problems

### ISSUE THREAD

**Issue by recepyazar** [mathis-s/SoomRV#36]
Hey,

Im trying baremetal asawmblies and also booting linux on your core.

After make command when i run dhry_1.s program:

```
~/riscv/SoomRV$ ./obj_dir/VTop test_programs/dhry_1.s
```
 i just see the output like this:
```
riscv32-unknown-linux-gnu-ld: warning: section `.data' type changed to PROGBITS
804 0
```

When i run make linux command output will be like this:
```
find: ‘/home/user/riscv/SoomRV/test_programs/linux/buildroot/output/target/usr/share/’: No such file or directory
find: ‘/home/user/riscv/SoomRV/test_programs/linux/buildroot/output/target/usr/libexec/’: No such file or directory
0+0 records in
0+0 records out
0 bytes copied, 4,6089e-05 s, 0,0 kB/s
make: *** [Makefile:91: linux] Error 1
```
when trying to run linux output will:
```
~/riscv/SoomRV$ ./obj_dir/VTop --device-tree=test_programs/linux/device_tree.dtb --backup-file=soomrv.backup test_programs/linux/linux_image.elf
826 0
```
my verilator version is:
```
Verilator 5.022 2024-02-24 rev v5.020-157-g2b4852048
```

How can i run linux or maybe C codes in baremetal? Are you running coremark in baremetal mode?

Thank you for your interest.

**Comment by mmxsrup** [comment]
I am having the same problem:
```
./obj_dir/VTop test_programs/hello_world.s
riscv32-unknown-linux-gnu-ld: warning: section `.data' type changed to PROGBITS
804 0
```

here is the version of Verilator:
```
Verilator 5.024 2024-04-05 rev v5.024
```

**Comment by mathis-s** [comment]
Hi guys, thanks for reporting the issue! This crash just occurs because I accidentally pushed some debugging code. For a quick fix, simply comment out this if statement:

https://github.com/mathis-s/SoomRV/blob/8b697b9399ada90d88f618891f6179ed058b598d/Top_tb.cpp#L1058-L1062

Unfortunately I can't work on SoomRV right now. I will properly fix this once I can get back to SoomRV in 3.5 weeks.
